### PR TITLE
Localization: Update French strings for `JSSidebar` KS macro

### PR DESCRIPTION
**Pull request by SphinxKnight**
## Summary
https://github.com/mdn/translated-content/issues/5135

### Problem

Some strings of this sidebar were not localized in `fr`

### Solution

Added fr strings / updated some others

## How did you test this change?

Spun up a local server

### CODE REVIEW

File: kumascript/macros/JsSidebar.ejs [hunk 1]
```diff
@@ -122,7 +122,7 @@ var text = mdn.localStringMap({
   },
   'fr': {
     'Overview': 'Aperçu des technologies JavaScript',
-    'Tutorials': 'Tutoriel\xa0:',
+    'Tutorials': 'Tutoriels&nbsp;:',
```
Comment (cw118) [inline kumascript/macros/JsSidebar.ejs:125]: The HTML entity doesn't seem to work here

```suggestion
    'Tutorials': 'Tutoriels\xa0:',
```

File: kumascript/macros/JsSidebar.ejs [hunk 2]
```diff
@@ -137,36 +137,36 @@ var text = mdn.localStringMap({
...
     'Equality': 'Différents tests d\'égalité',
-    'Closures': 'Closures',
+    'Closures': 'Fermetures (closures)',
     'Advanced': 'Avancé',
-    'Inheritance': 'Héritage et la chaîne de prototypes',
+    'Inheritance': 'L\'héritage et la chaîne de prototypes',
     'Strict_mode': 'Mode strict',
     'Typed_arrays': 'Tableaux typés en JavaScript',
     'Memory_Management': 'Gestion de la mémoire',
     'Event_Loop': 'Concurrence et boucle des événements',
```
Comment (cw118) [inline kumascript/macros/JsSidebar.ejs:161]:
```suggestion
    'Event_Loop': 'Concurrence et boucle des évènements',
```

File: kumascript/macros/JsSidebar.ejs [hunk 3]
```diff
@@ -137,36 +137,36 @@ var text = mdn.localStringMap({
...
-    'Reference': 'Références\xa0:',
+    'Reference': 'Références&nbsp;:',
```
Comment (cw118) [inline kumascript/macros/JsSidebar.ejs:162]: (cf. [the comment above]())

```suggestion
    'Reference': 'Références\xa0:',
```

File: kumascript/macros/JsSidebar.ejs [hunk 4]
```diff
@@ -137,36 +137,36 @@ var text = mdn.localStringMap({
...
     'Global_Objects': 'Objets natifs',
     'Operators': 'Expressions & opérateurs',
     'Statements': 'Instructions & déclarations',
```
Comment (cw118) [inline kumascript/macros/JsSidebar.ejs:165]: (selon [le guide stylistique](https://github.com/mozfr/besogne/wiki/Guide-stylistique-pour-la-traduction#règles-typographiques))

```suggestion
    'Operators': 'Expressions et opérateurs',
    'Statements': 'Instructions et déclarations',
```

File: kumascript/macros/JsSidebar.ejs [hunk 5]
```diff
@@ -137,36 +137,36 @@ var text = mdn.localStringMap({
...
     'Functions': 'Fonctions',
     'Classes': 'Classes',
-    'Errors': 'Errors',
-    'More': 'Plus',
+    'Errors': 'Erreurs',
+    'More': 'Divers',
     'Lexical_grammar': 'Grammaire lexicale',
     'Enumerability':'Rattachement des propriétés',
```
Comment (cw118) [inline kumascript/macros/JsSidebar.ejs:171]: And at line 175 (I can't suggest :/), maybe `'Template_strings': 'Gabarit de chaînes de caractères',` instead?

```suggestion
    'Enumerability': 'Rattachement des propriétés',
```

**Review by cw118 — CHANGES_REQUESTED** [review]
Thank you again @SphinxKnight, just a few suggestions :)

**Comment by SphinxKnight** [comment]
subsumed by #6213

**Pull request closed without merging**